lib/types/index.d.ts

```diff
@@ -1,98 +1,130 @@
+    }
 
-        hasDefinition (key: any): any;
+    // Package exports
+    export class ContainerBuilder {
+        constructor (containerReferenceAsService?: boolean, defaultDir?: string|null);
 
-        hasParameter (key: any): any;
+        readonly defaultDir: string;
+        readonly containerReferenceAsService: boolean;
+        readonly definitions: Map<string, Definition>;
+        readonly instanceManager: InstanceManager;
+        readonly extensions: Extension[];
+        readonly services: Map<string, any>;
+        frozen: boolean;
+        logger: Logger;
 
-        isSet (id: any): any;
+        addCompilerPass (compilerPass: any, type?: PassConfigHook, priority?: number): void;
 
-        register (id: any, ...args: any[]): any;
+        compile (): void;
 
-        registerExtension (extension: any): void;
+        findDefinition (key: string): Definition;
 
-        remove (id: any): void;
+        findTaggedServiceIds (name: string): Map<any, any>;
 
-        removeDefinition (key: any): any;
+        get<T = any> (id: string): T;
 
-        set (id: any, instance: any): void;
+        getDefinition (key: string): Definition;
 
-        setAlias (alias: any, id: any): void;
+        getParameter<T extends Parameter> (key: string): T;
 
-        setDefinition (id: any, definition: any): any;
+        has (key: string): boolean;
 
-        setParameter (key: any, value: any): void;
+        hasDefinition (key: string): boolean;
 
-    }
+        hasParameter (key: string): boolean;
 
-    export class Definition {
-        constructor (...args: any[]);
+        isSet (id: string): boolean;
+
+        register (id: string, object: any, args: Argument[]): Definition;
+
+        registerExtension (extension: Extension): void;
 
-        addArgument (argument: any): any;
+        remove (id: string): void;
 
-        addMethodCall (method: any, ...args: any[]): any;
+        removeDefinition (key: string): boolean;
 
-        addProperty (key: any, value: any): any;
+        set (id: string, instance: any): void;
 
-        addTag (name: any, ...args: any[]): any;
+        setAlias (alias: string, id: string): void;
 
-        isPublic (...args: any[]): any;
+        setDefinition (id: string, definition: Definition): Definition;
 
-        setFactory (Object: any, method: any): void;
+        setParameter (key: string, value: Parameter): void;
 
     }
 
-    export class JsFileLoader {
-        constructor (...args: any[]);
+    export class Definition {
+        constructor (object: any, args: Argument[]);
+
+        addArgument (argument: Argument): Definition;
+
+        addMethodCall (method: string, args: any[]): Definition;
+
+        addProperty (key: string, value: any): Definition;
+
+        addTag (name: string, attributes: Map<any, any>): Definition;
+
+        isPublic (bypassPublic?: boolean): boolean;
 
-        load (...args: any[]): void;
+        setFactory (Object: Reference|Object, method: string): void;
```

Comment:
Should I include PackageReference too? Or is it ok?

Reply:
nope! PackageReference ain't go in that way.